apps/condo/domains/banking/schema/BankSyncTask.js

```diff
@@ -153,18 +158,40 @@ const BankSyncTask = new GQLListSchema('BankSyncTask', {
             }
         },
         beforeChange: async ({ resolvedData, context, operation }) => {
-            if (operation === 'create' && get(resolvedData, 'options.type') === SBBOL) {
-                if (!resolvedData.account) {
-                    throw new GQLError(ACCOUNT_IS_REQUIRED, context)
+            if (operation === 'create') {
+                const type = get(resolvedData, 'options.type')
+                let integrationId
+                if (type === SBBOL) {
+                    integrationId = BANK_INTEGRATION_IDS.SBBOL
+                } else if (type === _1C_CLIENT_BANK_EXCHANGE) {
+                    integrationId = BANK_INTEGRATION_IDS['1CClientBankExchange']
+                }
+                const bankIntegrationOrganizationContext = await BankIntegrationOrganizationContext.getOne(context, {
+                    organization: { id: resolvedData.organization },
+                    integration: {
+                        id: integrationId,
+                    },
+                })
+                // After first execution of sync operations, a record for BankIntegrationOrganizationContext will be created automatically
+                // For integration with SBBOL it will be created in sync operation right after completed authorization flow
+                // For integration with 1C, it will be created in importBankTransactions worker
+                // We should inspect only existing record
+                if (bankIntegrationOrganizationContext && bankIntegrationOrganizationContext.enabled === false) {
+                    throw new GQLError(DISABLED_BANK_INTEGRATION_ORGANIZATION_CONTEXT, context)
                 }
-                const account = await BankAccount.getOne(context, { id: resolvedData.account })
-                const integration = get(account, 'integrationContext.integration.id')
-                if (!integration) {
-                    await BankAccount.update(context, account.id, {
-                        integrationContext: { connect: { id: BANK_INTEGRATION_IDS.SBBOL } },
-                        dv: 1,
-                        sender: { dv: 1, fingerprint: 'BankSyncTask' },
-                    })
+                if (type === SBBOL) {
+                    if (!resolvedData.account) {
+                        throw new GQLError(ACCOUNT_IS_REQUIRED, context)
```

Comment:
It might be better to add a note here about the reason why you can't validate account at the field level

Reply:
Added